### ISSUE TITLE
DocumentWatcher should check validity of window.activeTextEditor before accessing its fields

### ISSUE THREAD

**Issue by bobbrow** [editorconfig/editorconfig-vscode#155]
### Additional comments or steps to reproduce

Hi there, I was just testing out some unreleased code changes in the cpptools extension and I caught an exception coming from the editorconfig extension in the DocumentWatcher object.  This section of code accesses window.activeTextEditor.document without first checking if window.activeTextEditor is defined.

The case that I'm hitting is that our extension is upgrading and saving a file in the workspace which appears to trigger this callback:

```ts
		subscriptions.push(workspace.onWillSaveTextDocument(async e => {
			let selections: Selection[];
			if (window.activeTextEditor.document === e.document) {
				selections = window.activeTextEditor.selections;
			}
			const transformations = this.calculatePreSaveTransformations(e.document);
			e.waitUntil(transformations);
			if (selections) {
				transformations.then(() => {
					window.activeTextEditor.selections = selections;
				});
			}
		}));
```

**Comment by jedmao** [comment]
Are you saying that a simple:
```js
if (!e.document) {
    return;
}
```
at the beginning of the handler would fix the issue?

**Comment by bobbrow** [comment]
I was suggesting adding:

```ts
if (window.activeTextEditor === undefined) {
    return;
}
```

but checking e.document might work too.  I didn't check the value of e when the debugger broke in.

**Comment by jedmao** [comment]
If you could submit a PR I would be happy to merge it in, but I don't think I'll find the time to implement this anytime soon. I have a new baby coming any moment now. Things are crazy.

**Comment by jedmao** [comment]
Published as v0.9.4, thanks!
